### PR TITLE
Issue #4. Remove notifications from OS tray when they are read in the Inbox page of CM.

### PR DESCRIPTION
**Pull request by Jacob-Cuevas**
IMPORTANT: This PR should be merged once the PR #8 is approved.

## Description

This PR adds the following changes:
- Adds a new method that will be used to remove notifications from the OS tray by a `notID` or remove the notification grouper that was manually created.
- Reverts changes of the `clearNotification` method since it is used to remove all notifications from the OS tray, including the badge number displayed in the app icon.

## Related Issue

https://github.com/TransformativeMed/cordova-plugin-push/issues/4

## Motivation and Context

This method was created as part of the solution for issue #3 , since there must be a way to remove the notification grouper from the OS tray.

## How Has This Been Tested?

These changes were testing in a mobile device with the following specs:
- `16.5.1` iOS version.
- iPhone XR model.

## Screenshots (if appropriate):

- Remove a single notification from the OS tray when the user opens the reminder that was created using its data.

https://github.com/TransformativeMed/cordova-plugin-push/assets/12654157/c08a2324-f45a-49db-9a6e-a6fa4f0d5f71

- Remove the notification grouper from the OS tray when the reminder that was created from a single notification is opened.

https://github.com/TransformativeMed/cordova-plugin-push/assets/12654157/52ec3281-7637-4794-aee9-1a53534838b9

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/ios/AppDelegate+notification.m [hunk 1]
```diff
@@ -182,22 +253,254 @@ - (void)pushPluginOnApplicationDidBecomeActive:(NSNotification *)notification {
...
+    // Display the notification on the OS tray including the sound and badge number contained in the push notification.
+    completionHandler(UNNotificationPresentationOptionAlert | UNNotificationPresentationOptionSound | UNNotificationPresentationOptionBadge);
+}
+
+// Cordova Push Plugin issues #3. Get the data of the pressed notification (single or grouped notifications) to send it to the CORES Mobile app.
+// https://github.com/TransformativeMed/cordova-plugin-push/issues/3
+- (void) sendResponseDataOfPressedNotification: (NSDictionary *)userInfo : (PushPlugin *) pushHandler {
+    
+    // Check if the notification was sent by the APNS (triggered by the CORES API) or it is a local push notification created by us to let the user open all notifications ("open all" notification).
+    if ( [userInfo objectForKey:@"coresPayload"] ) {
```
Comment (ElliotNB) [inline src/ios/AppDelegate+notification.m:283]: Ditto from my other code review in pull request #8

File: src/ios/AppDelegate+notification.m [hunk 2]
```diff
@@ -182,22 +253,254 @@ - (void)pushPluginOnApplicationDidBecomeActive:(NSNotification *)notification {
...
+            // local notification created by us (the "open all" notification) to group these notifications when there are more than 1 in the tray.
+            for (int i=0; i<notificationsCount; i++){
+                
+                // Get data of the existing notification that is displayed in the Notification center.
+                UNNotification *existingNotification = [notifications objectAtIndex:i];
+                
+                NSDictionary *userInfo = existingNotification.request.content.userInfo;
+                
+                // Check if the current notification is the one received by the APNS or the "open all" local notification.
+                if ([userInfo objectForKey:@"coresPayload"]) {
```
Comment (ElliotNB) [inline src/ios/AppDelegate+notification.m:396]: Ditto

**Review by ElliotNB — CHANGES_REQUESTED** [review]
Can we chat tomorrow around 12PM Pacific about the approach here? I'm hoping that we can keep CORES Mobile logic out of the plugin component itself. But I need to understand 1. if that's possible and 2. if it's going to add to the complexity.

**Comment by Jacob-Cuevas** [comment]
Closing this PR since the PR #10 will move these changes to the `master` branch. Also, I had a conversation with Elliot about these changes and for now those will be introduced as they are in the branch.

**Pull request closed without merging**